src/app/core/services/notifications/app-server-rest-notification.service.ts

```diff
@@ -0,0 +1,135 @@
+  FCM_NOTIFICATION_CONTROLLER = 'fcm-notification-controller'
+
+  constructor(
+    public notifications: NotificationGeneratorService,
+    public storage: StorageService,
+    public schedule: ScheduleService,
+    public config: SubjectConfigService,
+    public firebase: Firebase,
+    public platform: Platform,
+    public logger: LogService,
+    public remoteConfig: RemoteConfigService,
+    public localization: LocalizationService,
+    private appServerService: AppServerService
+  ) {
+    super(storage, config, firebase, platform, logger, remoteConfig)
+  }
+
+  getSubjectDetails() {
+    return Promise.all([
+      this.appServerService.checkProjectAndSubjectExistElseCreate(),
+      this.config.getSourceID()
+    ]).then(([user, sourceId]) => Object.assign({}, user, { sourceId }))
+  }
+
+  publishAllNotifications(user, limit): Promise<any> {
+    return this.schedule.getTasks(TaskType.ALL).then(tasks => {
+      const fcmNotifications = this.notifications
+        .futureNotifications(tasks, limit)
+        .map(t => this.format(t, user))
+      this.logger.log('NOTIFICATIONS Scheduling FCM notifications')
+      this.logger.log(fcmNotifications)
+      return Promise.all(
+        fcmNotifications
+          .map(n => this.sendNotification(n, user.subjectId, user.projectId))
+          .concat([this.setLastNotificationUpdate()])
+      )
+    })
+  }
+
+  publishTestNotification(user): Promise<void> {
+    return this.sendNotification(
+      this.format(this.notifications.createTestNotification(), user),
+      user.subjectId,
+      user.projectId
+    )
+  }
+
+  cancelAllNotifications(user): Promise<any> {
+    return this.appServerService.getApiClient().then(apiClient =>
+      apiClient.apis[
+        this.FCM_NOTIFICATION_CONTROLLER
+      ].deleteNotificationsForUser({
+        subjectId: user.subjectId,
+        projectId: user.projectId
+      })
+    )
+  }
+
+  cancelSingleNotification(user, notificationId) {
+    return this.appServerService.getApiClient().then(apiClient =>
+      apiClient.apis[this.FCM_NOTIFICATION_CONTROLLER]
+        .deleteNotificationUsingProjectIdAndSubjectIdAndNotificationId({
+          subjectId: user.subjectId,
+          projectId: user.projectId,
+          id: notificationId
+        })
+        .then(() =>
+          console.log('Success cancelling notification ' + notificationId)
+        )
+    )
+  }
+
+  private sendNotification(notification, subjectId, projectId): Promise<any> {
+    return this.appServerService.getApiClient().then(apiClient =>
+      apiClient.apis[this.FCM_NOTIFICATION_CONTROLLER]
+        .addSingleNotification(
+          { projectId: projectId, subjectId: subjectId },
+          { requestBody: notification.notificationDto }
+        )
+        .then(res => {
+          notification.notification.id = res.body.id
+          return this.logger.log('Successfully sent! Updating notification Id')
```

Comment:
Is the id being persisted anywhere?

Reply:
Yes it's stored in the storage in the notification object.